Sources/Sentry/SentryThreadInspector.m

```diff
@@ -28,27 +64,89 @@ - (id)initWithStacktraceBuilder:(SentryStacktraceBuilder *)stacktraceBuilder
 {
     NSMutableArray<SentryThread *> *threads = [NSMutableArray new];
 
-    SentryCrashMC_NEW_CONTEXT(context);
-    [self.machineContextWrapper fillContextForCurrentThread:context];
+    @synchronized(self) {
+        SentryCrashMC_NEW_CONTEXT(context);
+        SentryCrashThread currentThread = sentrycrashthread_self();
 
-    int threadCount = [self.machineContextWrapper getThreadCount:context];
+        [self.machineContextWrapper fillContextForCurrentThread:context];
+        int threadCount = [self.machineContextWrapper getThreadCount:context];
 
-    for (int i = 0; i < threadCount; i++) {
-        SentryCrashThread thread = [self.machineContextWrapper getThread:context withIndex:i];
-        SentryThread *sentryThread = [[SentryThread alloc] initWithThreadId:@(i)];
+        for (int i = 0; i < threadCount; i++) {
+            SentryCrashThread thread = [self.machineContextWrapper getThread:context withIndex:i];
+            SentryThread *sentryThread = [[SentryThread alloc] initWithThreadId:@(i)];
 
-        sentryThread.name = [self getThreadName:thread];
+            sentryThread.name = [self getThreadName:thread];
 
-        sentryThread.crashed = @NO;
-        bool isCurrent = thread == sentrycrashthread_self();
-        sentryThread.current = @(isCurrent);
+            sentryThread.crashed = @NO;
+            bool isCurrent = thread == currentThread;
+            sentryThread.current = @(isCurrent);
 
-        // For now we can only retrieve the stack trace of the current thread.
-        if (isCurrent) {
-            sentryThread.stacktrace = [self.stacktraceBuilder buildStacktraceForCurrentThread];
+            if (isCurrent) {
+                sentryThread.stacktrace = [self.stacktraceBuilder buildStacktraceForCurrentThread];
+            }
+
+            // We need to make sure the main thread is always the first thread in the result
+            if ([self.machineContextWrapper isMainThread:thread])
+                [threads insertObject:sentryThread atIndex:0];
+            else
+                [threads addObject:sentryThread];
         }
+    }
 
-        [threads addObject:sentryThread];
+    return threads;
+}
+
+- (NSArray<SentryThread *> *)getCurrentThreadsWithStackTrace
```

Comment:
`m`: I think it would be vital to explain why we choose a different approach than above here.